nemo/collections/nlp/data/information_retrieval/gpt_embedding_dataset.py

```diff
@@ -47,6 +48,7 @@ def __init__(
         truncation_method: str = 'right',
         special_tokens: Optional[Mapping[str, str]] = None,  # special tokens, a dictory of {token_type: token}
         data_type: str = 'train',  # train, query or doc
+        num_hard_negatives: int = 4,  # number of hard negatives to use per query during training
```

Comment:
could we have this set via a configurable flag as welll?

Reply:
You can specify the number of hard negatives to use during training in your config. It is then passed to the dataset. I have made this change in the embedding model code.